vortex-cxx/cpp/include/vortex/scan.hpp

```diff
@@ -56,23 +56,23 @@ class ScanBuilder {
     ScanBuilder &operator=(const ScanBuilder &) = delete;
```

Comment:
I like the idea you mentioned in the other PR guarding against self move assignment.

```
ScanBuilder &operator=(ScanBuilder &&other) noexcept {
  if (this != &other) {
    impl_ = std::move(other.impl_);
  }
  return *this;
} 
```
Let's add this in addition to the other aspects mentioned. Ideally, the API would be impossible to misuse.